apollo-runtime/src/commonMain/kotlin/com/apollographql/apollo3/ApolloClient.kt

```diff
@@ -180,7 +180,7 @@ private constructor(
     private var wsProtocolFactory: WsProtocol.Factory? = null
     private var httpExposeErrorBody: Boolean? = null
     private var webSocketEngine: WebSocketEngine? = null
-    private var webSocketReconnectWhen: ((Throwable) -> Boolean)? = null
+    private var webSocketReconnectWhen: (suspend (Throwable, attempt: Long) -> Boolean)? = null
```

Comment:
Nitpick: an Int is probably enough?

Reply:
Having an int will be fine for almost all cases indeed.
Though, using a `Long` doesn't cost anything and helps avoid silent bugs for anyone trying an intensive usage (e.g. tests).
Moreover, I just mimicked kotlin flow's `retryWhen`
```kotlin
fun <T> Flow<T>.retryWhen(predicate: suspend FlowCollector<T>.(cause: Throwable, attempt: Long) -> Boolean): Flow<T>
```

Reply:
Fair enough! 👍